shell_integration/windows/OCUtil/RemotePathChecker.cpp

```diff
@@ -172,10 +176,13 @@ RemotePathChecker::~RemotePathChecker()
     CloseHandle(_newQueries);
 }
 
-vector<wstring> RemotePathChecker::WatchedDirectories()
+void RemotePathChecker::UpdateWatchedDirectories(vector<wstring>& localWatchedDirectories, unsigned& localVersion) const
 {
+    if (localVersion == _watchedDirectoriesVersion)
```

Comment:
Humm yeah it's safe but I could theoretically miss  the change notify callback if that `int` is still in the CPU cache. I'll put an atomic.